src/main/java/io/jenkins/plugins/cloudevents/EndPoint.java

```diff
@@ -0,0 +1,99 @@
+package io.jenkins.plugins.cloudevents;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.message.MessageWriter;
+import io.cloudevents.core.v1.CloudEventBuilder;
+import io.jenkins.plugins.cloudevents.model.JobModel;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static io.jenkins.plugins.cloudevents.CloudEventMessageWriter.createMessageWriter;
+
+/**
+ * Represents an Endpoint-object which stores the URL to which an event will be sent along with the type of the event.
+ */
+public class EndPoint {
```

Comment:
EndPoint is a bit ambiguous and don't know how I feel about the upper case "p". Instead of calling this an Endpoint, lets call it what it is a CloudeventsHTTPSink where you are able to give a URL for the sync and specify which events would be sent. Let this HTTPSinkConfiguration be implemented from an abstract class called CloudeventsSink. In the future we can use the same abstract class to create sink classes which handle different kinds of event streams.

Reply:
i really like that idea 
I mentioned refactoring this particular class in the meeting and I think this'll be a good starting implementation.